### ISSUE TITLE
"devel" CI builds are failing

### ISSUE THREAD

**Issue by pohly** [intel/pmem-csi#873]
Symptom is that the build on 1.20 times out, but the root cause is that skew testing is failing.

From  https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/rest/organizations/jenkins/pipelines/pmem-csi/branches/devel/runs/308/nodes/87/steps/110/log/?start=0

```
lvm-testing E2E [Testpattern: DynamicPV Filesystem {​​​​​​\"FSType\":\"\",\"Parameters\":null}​​​​​​] skew everything [Slow]
...

[2021-01-31T11:56:38.603Z] Jan 31 11:56:38.483: FAIL: create lvm-testing-0.8 PMEM-CSI deployment
[2021-01-31T11:56:38.603Z] Unexpected error:
[2021-01-31T11:56:38.603Z]     <*errors.errorString | 0xc004fe40e0>: {​​​​​​
[2021-01-31T11:56:38.603Z]         s: "command \"test/setup-deployment.sh\" failed: exit status 1\nOutput: INFO: deploying from /mnt/workspace/pmem-csi_devel/_work/pmem-csi-release-0.8/deploy/kubernetes-1.20/lvm/testing\nsecret/pmem-csi-registry-secrets configured\nsecret/pmem-csi-node-secrets configured\n/mnt/workspace/pmem-csi_devel/_work/pmem-csi-release-0.8/deploy/kustomize/lvm/testing is missing.\n",
[2021-01-31T11:56:38.603Z]     }​​​​​​
[2021-01-31T11:56:38.603Z]     command "test/setup-deployment.sh" failed: exit status 1
[2021-01-31T11:56:38.603Z]     Output: INFO: deploying from /mnt/workspace/pmem-csi_devel/_work/pmem-csi-release-0.8/deploy/kubernetes-1.20/lvm/testing
[2021-01-31T11:56:38.603Z]     secret/pmem-csi-registry-secrets configured
[2021-01-31T11:56:38.603Z]     secret/pmem-csi-node-secrets configured
[2021-01-31T11:56:38.603Z]     /mnt/workspace/pmem-csi_devel/_work/pmem-csi-release-0.8/deploy/kustomize/lvm/testing is missing.
[2021-01-31T11:56:38.603Z]     
[2021-01-31T11:56:38.603Z] occurred
```

**Comment by pohly** [comment]
Failing skew testing is getting fixed in PR #874 but then testing is still a lot slower. The reason for that is unknown, so to get our CI working again that PR disables testing on 1.20.

**Comment by pohly** [comment]
Here's an uncommitted patch that I was using locally to get some timing information for individual steps of a test:
```
diff --git a/test/e2e/operator/deployment_api.go b/test/e2e/operator/deployment_api.go
index db79c336a..ed0c7975f 100644
--- a/test/e2e/operator/deployment_api.go
+++ b/test/e2e/operator/deployment_api.go
@@ -555,7 +555,7 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 				},
 			}
 
-			By(fmt.Sprintf("Starting application pod '%s'", app.Name))
+			defer by(fmt.Sprintf("Starting application pod '%s'", app.Name))()
 			Eventually(func() error {
 				_, err := f.ClientSet.CoreV1().Pods(app.Namespace).Create(context.Background(), app, metav1.CreateOptions{})
 				deploy.LogError(err, "create pod %q error: %v, will retry...", app.Name, err)
@@ -566,7 +566,7 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 		}
 
 		stopPod := func(p *corev1.Pod) {
-			By(fmt.Sprintf("Stopping application pod '%s'", p.Name))
+			defer by(fmt.Sprintf("Stopping application pod '%s'", p.Name))
 			Eventually(func() error {
 				err := f.ClientSet.CoreV1().Pods(p.Namespace).Delete(context.Background(), p.Name, metav1.DeleteOptions{})
 				if err != nil && errors.IsNotFound(err) {
@@ -578,7 +578,7 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 		}
 
 		checkIfPodRunning := func(p *corev1.Pod) {
-			By(fmt.Sprintf("Ensure application pod '%s' is running", p.Name))
+			defer by(fmt.Sprintf("Ensure application pod '%s' is running", p.Name))
 			Eventually(func() error {
 				pod, err := f.ClientSet.CoreV1().Pods(p.Namespace).Get(context.Background(), p.Name, metav1.GetOptions{})
 				if err != nil {
@@ -635,12 +635,15 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 
 						deployment = deploy.CreateDeploymentCR(f, deployment)
 						defer deploy.DeleteDeploymentCR(f, deployment.Name)
-						deploy.WaitForPMEMDriver(c,
-							&deploy.Deployment{
-								Namespace:  d.Namespace,
-								DriverName: driverName,
-							})
-						validateDriver(deployment, true)
+						func() {
+							defer by("wait for driver")()
+							deploy.WaitForPMEMDriver(c,
+								&deploy.Deployment{
+									Namespace:  d.Namespace,
+									DriverName: driverName,
+								})
+							validateDriver(deployment, true)
+						}()
 
 						// NOTE(avalluri): As the current operator does not support deploying
 						// the driver in 'testing' mode, we cannot directely access CSI
@@ -657,15 +660,21 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 						defer deletePVC(f, pvc.Namespace, pvc.Name)
 
 						// Wait till a volume get provisioned for this claim
-						err := e2epv.WaitForPersistentVolumeClaimPhase(corev1.ClaimBound, f.ClientSet, pvc.Namespace, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
-						Expect(err).NotTo(HaveOccurred(), "Persistent volume claim bound failure")
+						func() {
+							defer by("wait for volume")()
+							err := e2epv.WaitForPersistentVolumeClaimPhase(corev1.ClaimBound, f.ClientSet, pvc.Namespace, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
+							Expect(err).NotTo(HaveOccurred(), "Persistent volume claim bound failure")
+						}()
 
-						pvc, err = f.ClientSet.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.Background(), pvc.Name, metav1.GetOptions{})
+						pvc, err := f.ClientSet.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.Background(), pvc.Name, metav1.GetOptions{})
 						Expect(err).NotTo(HaveOccurred(), "failed to get updated volume claim: %q", pvc.Name)
 						framework.Logf("PVC '%s', Volume Ref: %s", pvc.Name, pvc.Spec.VolumeName)
 
 						// Switch device mode
-						deployment = switchDeploymentMode(c, f, deployment.Name, d.Namespace, to)
+						func() {
+							defer by("switch device mode")()
+							deployment = switchDeploymentMode(c, f, deployment.Name, d.Namespace, to)
+						}()
 
 						postSwitch(from, to, driverName, pvc)
 
@@ -1236,7 +1245,8 @@ func createPVC(f *framework.Framework, namespace, name, storageClassName string)
 }
 
 func deletePVC(f *framework.Framework, namespace, name string) {
-	framework.Logf("Deleting PVC %q", name)
+	defer by(fmt.Sprintf("Deleting PVC %q", name))()
+
 	pvc, err := f.ClientSet.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), name, metav1.GetOptions{})
 	if err != nil {
 		ExpectWithOffset(1, errors.IsNotFound(err)).Should(BeTrue(), "Get PVC '%s'", name)
@@ -1270,3 +1280,21 @@ func deletePVC(f *framework.Framework, namespace, name string) {
 		return false
 	}, "3m", "1s").Should(BeTrue(), "Get PV '%s'", pvName)
 }
+
+func by(msg string) func() {
+	start := time.Now()
+	By("starting: " + msg)
+	return func() {
+		end := time.Now()
+		duration := end.Sub(start)
+		result := "successful"
+		r := recover()
+		if r != nil {
+			result = fmt.Sprintf("failed: %v", r)
+		}
+		By(fmt.Sprintf("done after %s: %s, %s", duration, msg, result))
+		if r != nil {
+			panic(r)
+		}
+	}
+}
```

I ran that with `make test_e2e TEST_E2E_FOCUS=olm.*API.*switch.device.mode.direct-to-lvm.use.volume.in.different.mode` and then got:
```
STEP: done after 14.123827067s: wait for driver, successful
STEP: done after 14.053438746s: wait for volume, successful
STEP: done after 30.256286389s: switch device mode, successful
STEP: done after 4.040943ms: Starting application pod 'switch-mode-app', successful
STEP: done after 37.2174584s: Deleting PVC "switch-mode-pvc", successful
STEP: done after 7.097911ms: Deleting PVC "switch-mode-pvc", successful
```

Initial results seemed to indicate that 1.20 was slower for some steps, but it wasn't always the case. Needs further work.

**Comment by pohly** [comment]
Resolved by taking testing of Kubernetes 1.20 out of the CI. We now need to bring it back (issue #886).
